cli/download_deno_binary.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use std::path::Path;
+use std::path::PathBuf;
+
+use crate::http_util::HttpClient;
+use crate::http_util::HttpClientProvider;
+use crate::util::progress_bar::ProgressBar;
+use crate::util::progress_bar::ProgressBarStyle;
+use deno_core::anyhow::bail;
+use deno_core::error::AnyError;
+
+use crate::cache::DenoDir;
+use crate::shared::ReleaseChannel;
+
+pub async fn download_deno_binary(
+  http_client_provider: &HttpClientProvider,
+  deno_dir: &DenoDir,
+  binary_kind: &str,
+  target: &str,
+  version_or_git_hash: &str,
+  release_channel: ReleaseChannel,
+) -> Result<PathBuf, AnyError> {
+  let binary_name = format!("{}-{}.zip", binary_kind, target);
+  let binary_path_suffix = match release_channel {
+    ReleaseChannel::Canary => {
+      format!("canary/{}/{}", version_or_git_hash, binary_name,)
+    }
+    _ => {
+      format!("release/v{}/{}", version_or_git_hash, binary_name)
+    }
+  };
+
+  let download_directory = deno_dir.dl_folder_path();
+  let binary_path = download_directory.join(&binary_path_suffix);
+
+  if !binary_path.exists() {
+    let http_client = http_client_provider.get_or_create()?;
+    download_base_binary(
+      &http_client,
+      &download_directory,
+      &binary_path_suffix,
+    )
+    .await?;
+  }
+
+  Ok(binary_path)
+}
+
+async fn download_base_binary(
+  http_client: &HttpClient,
+  output_directory: &Path,
+  binary_path_suffix: &str,
+) -> Result<(), AnyError> {
+  let download_url = format!("https://dl.deno.land/{binary_path_suffix}");
```

Comment:
That means we stop downloading `github.com/denoland/deno/releases` altogether?

Reply:
Yes